server/src/main/java/org/opensearch/action/search/TransportSearchAction.java

```diff
@@ -490,11 +428,15 @@ private void executeRequest(
             relativeStartNanos,
             System::nanoTime
         );
-
-        final List<SearchRequestOperationsListener> searchListenersList = createSearchListenerList(originalSearchRequest, timeProvider);
-        SearchRequestContext searchRequestContext = new SearchRequestContext(
-            new SearchRequestOperationsListener.CompositeListener(searchListenersList, logger)
-        );
+        final boolean phaseTookEnabled;
+        if (originalSearchRequest.isPhaseTook() == null) {
+            phaseTookEnabled = clusterService.getClusterSettings().get(SEARCH_PHASE_TOOK_ENABLED);
+        } else {
+            phaseTookEnabled = originalSearchRequest.isPhaseTook();
+        }
```

Comment:
Maybe we can update the originalSearchRequest with the phaseTookEnabled, and pass originalSearchRequest within SearchRequestContext. I guess that should keep the extensibility and establishes approach for adding flags similar to phaseTookEnabled in future.

Reply:
Updated the PR to include originalSearchRequest as part of the context object.